File: API/Entities/Message.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace API.Entities
{
    public class Message
    {
        public int Id { get; set; }
        public int SenderId { get; set; }
        public string SenderUserName { get; set; }
        public int ReceiverId { get; set; }
        public string ReceiverUserName { get; set; }
        public string Content { get; set; }
        public DateTime? DateRead { get; set; }
        public DateTime MessageSent { get; set; } = DateTime.Now;
        public bool SenderDeleted { get; set; } = false;
        public bool ReceiverDeleted { get; set; } = false;
        public bool IsRead { get; set; } = false;

        public Users Sender { get; set; }
        public Users Receiver { get; set; }
    }
}

File: Client/src/app/_models/message.ts
export interface IMessage {
  id: number;
  //send
  senderId: number;
  senderUserName: string;
  senderPhotoUrl: string;
  //receiver
  receiverId: number;
  receiverUserName: string;
  receiverPhotoUrl: string;
  //
  content: string;
  dateRead: Date;
  messageSent: Date;
  isRead: boolean;
}
export class MessageParams {
  container: string = 'Inbox'; //Outbox , Unread , Inbox
  pageNumber: number = 1;
  pageSize: number = 10;
}
